libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java

```diff
@@ -495,10 +496,10 @@ public QnAMakerRecognizer() {
             httpClient = this.httpClient;
         }
 
-        Pair endpointPair = this.endpointKey.tryGetValue(dc.getState());
-        Pair hostNamePair = this.hostName.tryGetValue(dc.getState());
-        Pair knowledgeBaseIdPair = this.knowledgeBaseId.tryGetValue(dc.getState());
-        Pair logPersonalInformationPair = this.logPersonalInformation.tryGetValue(dc.getState());
+        Pair endpointPair = new Pair(this.endpointKey, "");
+        Pair hostNamePair = new Pair(this.hostName, "");
+        Pair knowledgeBaseIdPair = new Pair(this.knowledgeBaseId, "");
+        Pair logPersonalInformationPair = new Pair(this.logPersonalInformation, null);
```

Comment:
I'm not quite sure about this, maybe we need to debug C# to understand which is the correct key/value (pair)

Reply:
In C# has: the value (endpointKey, hostName, etc) and a null value. I replaced it with a empty string to avoid null pointer exception

Reply:
I'd prefer the `null` as it will be simplier to validate it in the future, what do you think?

Reply:
Yes sure, I'll update it